corehq/apps/geospatial/models.py

```diff
@@ -81,3 +83,18 @@ def disbursement_solver(self):
         return self.VALID_DISBURSEMENT_ALGORITHM_CLASSES[
             self.selected_disbursement_algorithm
         ]
+
+    @property
+    def api_token(self):
+        if self._api_token and self._api_token.startswith(f'${ALGO_AES}$'):
+            ciphertext = self._api_token.split('$', 2)[2]
+            return b64_aes_decrypt(ciphertext)
+        return self._api_token
+
+    @api_token.setter
+    def api_token(self, value):
```

Comment:
[Here's](https://www.stavros.io/posts/how-replace-django-model-field-property/) also an article explaining what's happening here.

Baiscally I want to intercept a particular field's getter/setter methods, hence using this approach. And like Manish said, it's how we do it in other parts of the codebase as well.

Reply:
Thanks for the explanation, good to leave as is then. I just wanted to make sure there wouldn't be any confusion with sharing the same name.

Reply:
thanks for linking that @Charl1996 
It does seem we do a similar thing in the example I linked, with a slight difference

https://github.com/dimagi/commcare-hq/blob/b0f5643b36cf1ad4e4e66dfe458872d7278b9896/corehq/motech/models.py#L121-L132

Just to highlight the difference, is that the db column and the simultaneous property are encrypted and a separate property is added to handle the setter.
I can see only one advantage to that is that if anyone accesses `api_token`, which is something one would type when looking for `api_token` of the model object, one would always get the encrypted version. So for example, say in logs, it would be the encrypted token or while accidentally printing in console, or also in the view when rendering, avoiding accidental decryption and display of the token.
And thus forcing or being explicit that its the plaintext version of token being viewed at.

Even Django uses the same approach for its User model
```
class AbstractBaseUser(models.Model):
    password = models.CharField(_('password'), max_length=128)
```

So, if this was not a sensitive column, the implementation you shared would have been the right way to go about it but since this is a token we are probably better with keeping things explicit.

Let me know your thoughts on this.

Personally, I would explicitly decrypt only when necessary so the decrypted version is alive for the least time during code execution.